strapi/strapi
# feat(content-releases): Using useDocument for validations

This PR introduces `useDocument` as an unstable hook exported by the admin. 

⚠️ **Why use a hook and not duplicated validations in the Content-Releases plugin?:** So we don't have duplicate logic and two things to care. Right now, all the validation logic is on the `createYupSchema` function, the idea of this PR is create a new `useDocument` hook that we can shape in the future with more utils. 

⚠️ This is a first version of what we would like to have for a `useDocument` hook. The main problem right now as mention on the hook description is that contentManager store is empty if you are outside the content-manager, this means that you don't have any information about content type schemas or components outside the content-manager. For that reason, this first version of the hook is receiving the Content Type schema as a params instead of retrieving it for the store.

### How to test it:
- Create a new release and add an entry which is not valid.
- You should see the validations messages in the column
